src/rust/engine/rule_graph/src/lib.rs

```diff
@@ -633,11 +633,15 @@ impl<R: Rule> RuleGraph<R> {
       .collect();
     msgs.sort();
 
-    Err(format!(
-      "Rules with errors: {}\n\n  {}",
-      msgs.len(),
-      msgs.join("\n\n  ")
-    ))
+    if msgs.is_empty() {
+      Ok(())
+    } else {
+      Err(format!(
+        "Rules with errors: {}\n\n  {}",
+        msgs.len(),
+        msgs.join("\n\n  ")
+      ))
+    }
   }
```

Comment:
Using `@_uncacheable_rule` with `changed_files()` in this PR would fail with `Exception: Rules with errors: 0` without this change.

Reply:
Hm... fishy. Would you mind opening a ticket about that?